core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -203,6 +232,43 @@ public PGConnectionContext(CairoEngine engine, PGWireConfiguration configuration
         final int blockCount = enableInsertCache ? configuration.getInsertCacheBlockCount() : 1; // 8
         final int rowCount = enableInsertCache ? configuration.getInsertCacheRowCount() : 1; // 8
         this.typesAndInsertCache = new AssociativeCache<>(blockCount, rowCount);
+        this.batchCallback = new PGConnectionBatchCallback();
+        this.bindSelectColumnFormats = new IntList();
+    }
+
+    class PGConnectionBatchCallback implements BatchCallback {
+        @Override
+        public void preCompile(SqlCompiler compiler) throws SqlException {
+            prepareForNewBatchQuery();
+            PGConnectionContext.this.typesAndInsert = null;
```

Comment:
The line above, a call to prepareForNextBatchQuery, already sets `typesAndInsert` to null, should it also set `typesAndSelect` to null?, and then this line and the one bellow can go.

Reply:
Yes, both need to be null-ed. I moved these two lines from earlier version of compileQuery method to make sure it happens .